### PR TITLE
e2e: improvements

### PR DESCRIPTION
**Pull request by dougsland**
- e2e: use kubectl --all-namespaces
- e2e: specify image registry for kindest/node
- e2e: replace --timeout to --request-timeout
- e2e: show error if kind cannot load image
- e2e: fail if kind create cluster fails

### CODE REVIEW

**Comment by dougsland** [comment]
> @dougsland: Cannot trigger testing until a trusted user reviews the PR and leaves an
> `/ok-to-test` message.

Ok, I can't trigger /ok-to-test but I can't see how the patches are related to this issue.

File: hack/test_e2e.sh [hunk 1]
```diff
@@ -23,6 +23,9 @@ shopt -s expand_aliases
 OS=$(uname| tr '[:upper:]' '[:lower:]')
 CONTAINER_ENGINE="docker"
 
+# Users can specify docker.io, quay.io registry
+KINDEST_NODE_IMAGE="docker.io/kindest/node"
```
Comment (anusha94) [inline hack/test_e2e.sh:27]: Nice! Should we also allow this to be a command line arg with default set to docker.io? What do you think?

File: hack/test_e2e.sh [hunk 2]
```diff
@@ -282,7 +289,10 @@ function install_kpng {
     # TODO move this to ci:
     # docker load --input kpng-image.tar
     echo "loading kpng:test docker image"
-    kind load docker-image kpng:test --name ${E2E_CLUSTER_NAME}
+    CMD_KIND_LOAD_KPNG_TEST_IMAGE=("kind load docker-image kpng:test --name ${E2E_CLUSTER_NAME}")
+    ${CMD_BUILD_IMAGE}
```
Comment (anusha94) [inline hack/test_e2e.sh:293]: why do we need `CMD_BUILD_IMAGE` here? If I understand correctly, that's part of `docker_build` function right?
https://github.com/kubernetes-sigs/kpng/blob/8496fd72894dc6e887e8d0242f8cb6ba32572b6b/hack/test_e2e.sh#L86-L100

File: hack/test_e2e.sh (hunk 1 above)
Comment (dougsland) [reply hack/test_e2e.sh:27]: I agree but in the first place I would ship as static. In the next weeks we can add such feature as well.

File: hack/test_e2e.sh (hunk 2 above)
Comment (dougsland) [reply hack/test_e2e.sh:293]: Good catch, probably a buffer in my copy/past. Thanks!

**Comment by jayunit100** [comment]
should we merge it ?

**Comment by dougsland** [comment]
> should we merge it ?

Yes, If you folks agree of course :)

**Comment by dougsland** [comment]
@anusha94 @jayunit100 I believe I am fine with the test_e2e.sh suggestions now. If the patch got approved and merged I am going to switch context and start looking into the tests with Petr as discussed in the last meeting. Happy Holidays!

**Comment by dougsland** [comment]
The failures are not related to the patch but there are tests indeed not ready/green yet.

**Review by anusha94 — APPROVED** [review]
/lgtm

**Comment by dougsland** [comment]
@anusha94 could you please help force merge as the jobs are failing not related to the patch? Thanks

**Comment by anusha94** [comment]
> @anusha94 could you please help force merge as the jobs are failing not related to the patch? Thanks

@dougsland I don't have merge rights. Maybe @jayunit100 or @knabben can merge this?

**Pull request merged**